### PR TITLE
GH-108614: Remove non-debug uses of `#if TIER_ONE` and `#if TIER_TWO` from `_POP_FRAME` op.

### PR DESCRIPTION
**Pull request by markshannon**
* Issue: gh-108614

### CODE REVIEW

**Comment by gvanrossum** [comment]
Please give me some time to review this.

File: Python/ceval_macros.h [hunk 1]
```diff
@@ -373,3 +373,48 @@ static inline int _Py_EnterRecursivePy(PyThreadState *tstate) {
...
+
+
+/* Implementation of "macros" that modify the instruction pointer,
+ * stack pointer, or frame pointer.
+ * These need to treated differently by tier 1 and 2. */
+
+#if TIER_ONE
+
+#define STORE_IP() \
+do {frame->prev_instr = next_instr-1; } while (0)
```
Comment (gvanrossum) [inline Python/ceval_macros.h]: I cannot find any place that uses STORE_IP()?

**Review by gvanrossum — APPROVED** [review]
I'm guessing STORE_IP() is meant to be used by _PUSH_FRAME and you're leaving that for a follow-up PR?

File: Python/ceval_macros.h (hunk 1 above)
Comment (markshannon) [reply Python/ceval_macros.h]: I added all four, then used them where necessary.
I'll remove `STORE_IP` for now, but we will need it for `YIELD_VALUE`.

**Pull request merged**